### PR TITLE
CPLAT-4789: fix string refs

### PR DESCRIPTION
**Pull request by kealjones-wk**
## Issue
String refs were breaking due to a bundling issue in webpack when using 2 seperate files that would require('react'). Each file would use a different instance of react.

## Solution
Update webpack config to use the window.React instance for any require('react') that are not the initial provider of react.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -47,19 +47,23 @@ function createExports(exportMappings) {
...
+        },
+        entry: path.resolve(inputPath, entryFilename),
+        plugins: isProduction ? prodPlugins : devPlugins,
+        mode: isProduction ? "production" : "development",
+        externals: [{ window: "window" }],
+        devtool: "source-map",
+      };
+
+      if ( entryFilename != 'react.js' && !entryFilename.includes('react_with') ) {
+        exportObject.externals[0].react = "window.React";
```
Comment (greglittlefield-wf) [inline webpack.config.js]: Could you add a comment saying why this is necessary?

File: webpack.config.js [hunk 2]
```diff
@@ -47,19 +47,23 @@ function createExports(exportMappings) {
...
+          filename: outputFilename
+        },
+        entry: path.resolve(inputPath, entryFilename),
+        plugins: isProduction ? prodPlugins : devPlugins,
+        mode: isProduction ? "production" : "development",
+        externals: [{ window: "window" }],
+        devtool: "source-map",
+      };
+
+      if ( entryFilename != 'react.js' && !entryFilename.includes('react_with') ) {
```
Comment (greglittlefield-wf) [inline webpack.config.js]: #nit could we split this out to a variable?
```js
includeReact = entryFilename === 'react.js' || entryFilename.includes('react_with');

File: webpack.config.js (hunk 1 above)
Comment (kealjones-wk) [reply webpack.config.js]: I added a comment but i am not so great with the word so feel free to suggest something better :)

Comment (greglittlefield-wf) [reply webpack.config.js]: LGTM

**Pull request merged**